### ISSUE TITLE
Figure out whether the legacy API parts behind auth are still used (DELETE / POST / PUT)

### ISSUE THREAD

**Issue by diox** [mozilla/addons-server#1715]
Tracker: #1680.

The legacy API can do authentication and supports  `POST`, `PUT`, and `DELETE` operations on add-ons and versions. This is undocumented however. Is still still used ? This issue is about finding out the answer to that question :-)

**Comment by diox** [comment]
So, I played a bit with datadog. 

I made [a simple dashboard showing the `POST`, `PUT` and `DELETE` requests across the whole site](https://app.datadoghq.com/dash/100104), and we can see that we have a few POSTS, fewer PUTs, and only a couple DELETEs. That was across the whole website though, and this obviously includes a lot more than just the API - this just proves that we record the stats correctly.

Trying to show only the legacy API views... Well, these don't even show up in datadog metrics. We can see a very low count for `addons_prod.view.olympia.api.urls.SwitchToDRF.GET`, but the metrics for the other HTTP verbs don't even show up, so I suspect there are simply no calls at all!

In addition, I was also unable to find any calls for `authorize` or `oauth` or `request_token_ready` in `addons_prod`.

My conclusion is that both the authentication and the DELETE/POST/PUT handling is completely unused in the legacy API and can be removed. Can anyone else double-check my findings ?
